spark/v3.3/build.gradle

```diff
@@ -193,8 +193,12 @@ project(":iceberg-spark:iceberg-spark-runtime-${sparkMajorVersion}_${scalaVersio
 
   dependencies {
     api project(':iceberg-api')
-    implementation project(":iceberg-spark:iceberg-spark-${sparkMajorVersion}_${scalaVersion}")
-    implementation project(":iceberg-spark:iceberg-spark-extensions-${sparkMajorVersion}_${scalaVersion}")
+    implementation(project(":iceberg-spark:iceberg-spark-${sparkMajorVersion}_${scalaVersion}")) {
+      exclude group: 'org.scala-lang', module: 'scala-library'
```

Comment:
My previous comment may be unclear, my suggestion is to add the following part after line 190
```
      exclude group: 'org.scala-lang'
      exclude group: 'org.scala-lang.modules'
```

Reply:
sounds good to me as spark is providing the `scala.collection.compat` also.

https://files.slack.com/files-pri/T025T8UC953-F042D9QPEMA/image.png

Updated the PR. Manually verified that there is no scala folder with the runtime jars. let's wait for the build.